File: tests/Mug/fib/fib.c
int fib(int n) {
  int last = 1;
  int prelast = 0;
  while (n >= 0) {
    const int tmp = last + prelast;
    prelast = last;
    last = tmp;
    n--;
  }

  return last;
}

int main() {
  int result = 0;

  for (int i = 0; i <= 40; i++)
    result += fib(i);

  return result;
}

File: tests/Mug/fib/fib.zig
pub extern fn ExitProcess(code: c_uint) noreturn;

fn fib(n: i32) i32 {
  var last: i32 = 1;
  var prelast: i32 = 0;
  var i: i32 = n;
  while (i >= 0) {
    const tmp = last + prelast;
    prelast = last;
    last = tmp;
    i -= 1;
  }

  return last;
}

pub fn main() void {
  var result: i32 = 0;
  var i: i32 = 0;

  while (i <= 40) {
    result += fib(i);
    i += 1;
  }
  
  ExitProcess(@bitCast(c_uint, result));
}
